File: CountriesApp/app/src/main/java/hootor/com/countriesapp/adapter/CountryAdapter.java
package hootor.com.countriesapp.adapter;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.ArrayList;

import hootor.com.countriesapp.R;
import hootor.com.countriesapp.model.Result;

public class CountryAdapter extends RecyclerView.Adapter<CountryAdapter.CounrtyViewHolder> {

    private ArrayList<Result> countryList;

    public CountryAdapter(ArrayList<Result> countryList) {
        this.countryList = countryList;
    }

    @NonNull
    @Override
    public CounrtyViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {

        LayoutInflater layoutInflater = LayoutInflater.from(viewGroup.getContext());

        View view = layoutInflater.inflate(R.layout.row_country,viewGroup, false);

        return new CounrtyViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull CounrtyViewHolder counrtyViewHolder, int i) {

        Result result = countryList.get(i);

        counrtyViewHolder.countryNameTextView.setText(result.getName());

    }

    @Override
    public int getItemCount() {
        return countryList.size();
    }

    class CounrtyViewHolder extends RecyclerView.ViewHolder {

        TextView countryNameTextView;

        public CounrtyViewHolder(@NonNull View itemView) {
            super(itemView);
            countryNameTextView = itemView.findViewById(R.id.tv_country_name);
        }
    }
}

File: RetrofitDemo/app/src/main/java/com/android/hootr/retrofitdemo/adapters/CountryAdapter.kt
package com.android.hootr.retrofitdemo.adapters

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.android.hootr.retrofitdemo.R
import com.android.hootr.retrofitdemo.model.Result
import kotlinx.android.synthetic.main.country_item.view.*

class CountryAdapter() :
    RecyclerView.Adapter<CountryAdapter.CountryViewHolder>() {

    var resultArrayList: ArrayList<Result> = arrayListOf()
    set(value) {
        field = value
        notifyDataSetChanged()
    }
    inner class CountryViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        var countryNameTextView: TextView? = null

        init {
            countryNameTextView = itemView.countryNameTextView
        }

    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CountryViewHolder {
        val view: View =
            LayoutInflater.from(parent.context).inflate(R.layout.country_item, parent, false)
        return CountryViewHolder(view)
    }

    override fun getItemCount() = resultArrayList.size

    override fun onBindViewHolder(holder: CountryViewHolder, position: Int) {
        holder.countryNameTextView?.text = resultArrayList.get(position).name
    }

}